projects/ng-dynamic-component/src/lib/io/io.service.ts

```diff
@@ -74,6 +74,7 @@ export class IoService implements OnDestroy {
     private cfr: ComponentFactoryResolver,
     @Inject(EventArgumentToken)
     private eventArgument: string,
+    private cdr?: ChangeDetectorRef,
```

Comment:
Please also remove question mark as it's not optional.

Reply:
done. I needed to change IoFactoryService to make this work. IoFactorySerivce is now provided by DynamicDirectivesDirective. Technically a breaking change, but also a bugfix. This is overall more correct, because the previous implementation always used the root Injector. Now it uses the Injector from where DynamicDirectivesDirective is instantiated. If somewhere in the injector tree a new value for e.g. EventArgumentToken is provided, that's now correctly picked up by the service.

Reply:
Do not worry about the factory - it's a private api so cannot be a breaking change =)